### ISSUE TITLE
Firefox 38 will not allow (maybe just warn for now) unsigned add-ons

### ISSUE THREAD

**Issue by Yonezpt** [YePpHa/YouTubeCenter#1676]
https://blog.mozilla.org/addons/2015/04/15/the-case-for-extension-signing/

With Firefox 38 to be released on May 12th, add-ons that are not signed in by Mozilla will no longer be able to install on stable Firefox versions, or maybe they will only post a warning during this transition (I am confused about this so I can't be sure about which one), so it is expected that either then or in the near future the YTC add-on will simply stop working if it doesn't get signed.

The solutions then will be:

 - Not update Firefox past v.37
 - Sign the addon
 - Use a non-signed requirement browser version, such as Nightly if I recall correctly
 - Use the userscript version and hope that Greasemonkey also signs their addon

@YePpHa I am posting this in case you didn't know already and for anyone else that might be interested in such information.

**Comment by MysteryIII** [comment]
It can be "fixed" in about:config
Set 'extensions.install.requireSecureOrigin' to 'false'.

**Comment by Zarggg** [comment]
Not quite. That preference actually [allows the installation of addons from web pages using HTTP](https://blog.mozilla.org/addons/2015/03/13/domain-whitelisting-disabled-non-https/) (as opposed to HTTPS).

The blog post referenced by @Yonezpt refers to extensions requiring a signed security certificate in order to be installed regardless of the source.

**Comment by Yonezpt** [comment]
Exactly, any and all add-ons that are not signed by Mozilla won't be allowed to install. From further reading it seems that they will just warn users from 38+ until 12 weeks pass, then they will start blocking unsigned add-ons.

**Comment by SammysHP** [comment]
In some weeks we cannot install the developer build in Firefox because the XPI here is not signed. Please sign your XPI as fast as possible.

**Comment by Yonezpt** [comment]
@SammysHP This is due to a recent update, but apart from stable and beta versions, all other Firefox releases can run unsigned add-ons, you just have to disable the forced signing via the `about:config` page, locate the setting `xpinstall.signatures.required` and change it to `false`

**Comment by SammysHP** [comment]
@Yonezpt But I want to keep my stable release version for daily usage.

**Comment by Yonezpt** [comment]
Then you will have to use the userscript version of YTC because the add-on will be disabled once 42 stable is released unless it is signed.

**Comment by SammysHP** [comment]
Where's the problem to sign the developer build XPI? Mozilla says it's fast and easy.

**Comment by Yonezpt** [comment]
Read https://github.com/YePpHa/YouTubeCenter/issues/1325#issuecomment-68047353
